src/ProjectOrigin.WalletSystem.IntegrationTests/Repositories/CertificateRepositoryTests.cs

```diff
@@ -166,14 +168,168 @@ public async Task GetAllOwnedCertificates()
         await _repository.InsertWalletSlice(slice3);
         await _repository.InsertWalletSlice(sliceWithDifferentOwner);
 
-        var certificates = await _repository.GetAllOwnedCertificates(owner1, new CertificatesFilter());
+        var certificates = await _repository.GetAllOwnedCertificates(owner1, new CertificatesFilter(SliceState.Available));
 
         certificates.Should().HaveCount(2).And.Satisfy(
             c => c.Id == certificate1.Id && c.Slices.Sum(x => x.Quantity) == slice1.Quantity + slice2.Quantity,
             c => c.Id == certificate2.Id && c.Slices.Sum(x => x.Quantity) == slice3.Quantity
         );
     }
 
+    [Fact]
+    public async Task GetAllOwnedCertificates_WhenGettingTotalCertificates()
+    {
```

Comment:
Where are the Integration tests, where you test the API endpoints?

Reply:
There are no integration tests on these endpoints. My best bet is that it is because we have all the logic in the repository